### PR TITLE
Drones are now cargo-purchaseable, makes the console public

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

I've thrown around a lot of ideas for maint drones as an alternative to #24207. I think this is the best one.
- Drones are no longer infinitely spawnable from the drone fab
- Drone fabs are now "drone storages"
- Drones are ordered from cargo for 400 credits for 3 of them
- Drones no longer gib on death
- Emagged drones now gib at the end of their timer instead of just shutting down
- Anyone with Maintenance access can shut down a maint drone by swiping their ID in it (as long as its not emagged)
- Drone storage now displays number of active drones (players) and stored drones (clientless drones that players can join as)
- Drone storages can now be found in maintenance instead of engineering. They really are **MAINTENANCE** drones.
- Drones no longer have a max-cap of 5 per fabricator.

## Why It's Good For The Game

This PR aims to keep the drone/dchat supporting role, but in a way that requires player agency. 

## Images of changes

image
image
image

## Testing

Ordered some drones, tested shit with em

## Changelog
:cl:
tweak: Drones are now cargo-orderable instead of being in infinite supply from the drone fabricator
/:cl:

### CODE REVIEW

**Comment by Daylight2** [comment]
Since they're now in limited supply, I feel like drones should pathfind back to the fab and become available to other players if you ghost out of them. Also some tweak to either their capability or lawset might be warranted, since they're no longer free and won't show up every shift.

**Review by 1080pCat — COMMENTED** [review]
I believe you've reverted funnymans request console update pr.

Also could you move the drone fab on meta down to around here?
image
where it is now looks.. outa place

**Comment by FunnyMan3595** [comment]
Neat idea, but it doesn't solve the admin work or bad lawset issues that plague drones.

**Comment by Contrabang** [comment]
> Neat idea, but it doesn't solve the admin work or bad lawset issues that plague drones.

I think that limiting the supply of them is something that is needed for the lawset to be reworked. As for the admin issues, I do not have enough experience admining to say anything on that, nor where to go with that.

**Comment by Contrabang** [comment]
> I've got.. Quite a strong opinion against this, i've been thinking about it for a while.
> 
> I do not believe this would solve anything. Making the drones cost 400 credits for 3 drones, is just, weird. I expect this to result in the dronefab being neglected, and just never stocked.
> 
> The primary issue with drones is their horrid laws, that contradict themselves, and are super vague. Not that they're easily accessible. Moving the drones fabs to maintenance is.. interesting, im not super apposed to it, but im concerned about tiders seeing the shutdown button and just shutting down drones for the hell of it.
> 
> Also with how.. short the "why its good for the game" is, it seems like even you dont have super solid reasoning for it, somewhat feels like a change to change it. No offense.
> 
> TLDR: The drones need new laws, not mechanical changes.

Fair, this idea started when the 2 PRs open for drones were
1. Removing them from the game completely
2. Making them into a pAI modsuit module
I'm fine with this PR being closed if there are enough objections against it

**Comment by bruhlookatthisdood** [comment]
is there any chance the round can start with like, one drone or even two?
i don't want to be locked out of the role if cargo just completely forgets or refuses to buy them.

**Comment by Kenionatus** [comment]
I feel like the drone fab/storage being in engineering would be even more important with this PR, because then engineering would be clearly responsible for stocking it.

**Review by Spaghetti-bit — APPROVED** [review]
approve for code

File: code/modules/mob/living/silicon/robot/drone/drone_console.dm [hunk 1]
```diff
@@ -44,18 +44,27 @@
...
-		data["drone_prod"] = dronefab.produce_drones
-		data["drone_progress"] = dronefab.drone_progress
+
 	data["selected_area"] = drone_call_area
 	data["ping_cd"] = ping_cooldown > world.time ? TRUE : FALSE
 
 	data["drones"] = list()
+	var/active_drones = 0
+	var/stored_drones = 0
+	var/total_drones = 0
```
[the remark this answers is not in the document]
Comment (Contrabang) [reply code/modules/mob/living/silicon/robot/drone/drone_console.dm:60]: Theres no use for auto-doc comments in a proc, they won't be recognized. Normal comments are fine for situations like these, but I feel these are already self-documenting enough

**Comment by SteelSlayer** [comment]
This PR has failed the Design team vote with 3 against and 0 in favor. Here are the objection reasons:

**Member 1:** Neat idea, but it doesn't solve the admin work or bad lawset issues that plague drones.
**Member 2:** Same thoughts as Member 1. Still does not solve the drone issue and them being how they exist currently on station with their laws, admin issues, and in general bad concept.
**Member 3:** Yeah, while it is nice that something is attempting to be done to help the drones along, this doesn't seem like it'll do a whole lot to fix the issues that people have with them.
